numpyro/distributions/transforms.py

```diff
@@ -451,6 +452,36 @@ def log_abs_det_jacobian(self, x, y, intermediates=None):
         return x
 
 
+class SoftplusTransform(Transform):
+    r"""
+    Transform from unconstrained space to positive domain via softplus :math:`y = \log(1 + \exp(x))`.
+    The inverse is computed as :math:`y = \log(\exp(x) - 1)`.
+    """
+    def __init__(self):
+        self.domain = constraints.real
```

Comment:
I think you can simply define default attributes
```
class SoftplusTransform(Transform):
    domain = constraints.real
    codomain = constraints.positive
```
Otherwise, it is better to have the `domain` argument in `__init__` (as in your initial commit).

Reply:
Changed to a simpler case.